server/src/main/java/com/genymobile/scrcpy/control/UhidManager.java

```diff
@@ -52,13 +82,15 @@ public void open(int id, int vendorId, int productId, String name, byte[] report
         try {
             FileDescriptor fd = Os.open("/dev/uhid", OsConstants.O_RDWR, 0);
             try {
-                FileDescriptor old = fds.put(id, fd);
+                // Must be unique across the system
+                String inputPort = "scrcpy:" + Os.getpid() + ":" + id;
```

Comment:
There is only one virtual display per scrcpy session, do we need one input port per UHID device?

I think we could just use `String inputPort = "scrcpy:" + Os.getpid();` and call `addUniqueIdAssociationByPort()` only once (and all UHID devices would use the same input port associated to the single display id).

I'm not asking you to make any change, I would just like to know what you think about it.

(Currently, I cannot test virtual displays because of #5523 on my current device.)

Reply:
I never thought about that. Initially I saw Android Studio is using different `phys` values for each device: https://cs.android.com/android-studio/platform/tools/adt/idea/+/mirror-goog-studio-main:streaming/screen-sharing-agent/app/src/main/cpp/virtual_input_device.cc;l=66-68;drc=fc06f05ca98ff7d3c8558976fd7eaf1b5480e2e8

But I believe multiple devices can have the same `phys` value, it means they are in one physical device, but with multiple input features. And `addUniqueIdAssociationByPort` still works.

The original commit has been backed up at https://github.com/yume-chan/scrcpy/tree/feat/associate-inpt-port-2